pact/policy-manager/policy-manager.pact

```diff
@@ -0,0 +1,339 @@
+(namespace (read-msg 'ns))
+
+(module policy-manager GOVERNANCE
+
+  (defcap GOVERNANCE ()
+    (enforce-guard 'marmalade-admin ))
+
+  (use kip.token-policy-v2 [token-info])
+  (use marmalade.quote-manager)
+  (use marmalade.quote-manager [quote-spec quote-msg fungible-account])
+
+  (defconst QUOTE-MSG-KEY "quote"
+    @doc "Payload field for quote spec")
+
+  (defconst UPDATE-QUOTE-PRICE-MSG-KEY "update_quote_price"
+    @doc "Payload field for quote spec")
+
+  (defconst BUYER-FUNGIBLE-ACCOUNT-MSG-KEY "buyer_fungible_account"
+    @doc "Payload field for buyer's fungible account")
+
+  (defcap POLICY_MANAGER:bool ()
+    @doc "Ledger module guard for policies to be able to validate access to policy operations."
+    true
+  )
+
+  (defcap ESCROW (sale-id:string)
+    @doc "Capability to be used as escrow's capability guard"
+    true
+  )
+
+  (defun get-escrow-account:object{fungible-account} (sale-id:string)
+    { 'account: (create-principal (create-capability-guard (ESCROW sale-id)))
+    , 'guard: (create-capability-guard (ESCROW sale-id))
+    })
+
+  (defun policy-manager-guard:guard ()
+    (create-capability-guard (POLICY_MANAGER))
+  )
+
+  ;; Saves ledger guard information
+  (defschema ledger
+    ledger-guard:guard
+  )
+
+  (deftable ledgers:{ledger}
+    @doc "Designed to save one ledger in a single row")
+
+  (defun enforce-ledger:bool ()
+    @doc "Enforces that function is called from the saved ledger"
+    (with-read ledgers "" {
+      "ledger-guard":= ledger-guard
+      }
+      (enforce-guard ledger-guard)
+    )
+  )
+
+  (defun init:bool(ledger-guard:guard)
+    @doc "Must be initiated with ledger information"
+    (with-capability (GOVERNANCE)
+      (insert ledgers "" {
+        "ledger-guard": ledger-guard
+      })
+    )
+    true
+  )
+
+  ;; Saves Concrete policy information
+  (defschema concrete-policy
+    policy:module{kip.token-policy-v2}
+  )
+
+  (defcap CONCRETE_POLICY:bool (policy-field:string policy:module{kip.token-policy-v2})
+    @doc "Event emission purpose"
+    @event
+    true
+  )
+
+  (deftable concrete-policies:{concrete-policy})
+
+  (defconst NON_FUNGIBLE_POLICY:string 'non-fungible-policy )
+  (defconst ROYALTY_POLICY:string 'royalty-policy )
+  (defconst COLLECTION_POLICY:string 'collection-policy )
+  (defconst GUARD_POLICY:string 'guard-policy )
+  (defconst CONCRETE_POLICY_LIST:[string]
+    [NON_FUNGIBLE_POLICY ROYALTY_POLICY COLLECTION_POLICY GUARD_POLICY] )
+
+  (defun write-concrete-policy:bool (policy-field:string policy:module{kip.token-policy-v2})
```

Comment:
I only see this in repl files. How is this installed? I don't see it in yaml files.

Reply:
I've added [pact/policy-manager/manager-init.pact](https://github.com/kadena-io/marmalade/blob/4bf65aa1646a6038ba634f814a789e868efc5ae4/pact/policy-manager/manager-init.pact)